parcels/particlesets/baseparticleset.py

```diff
@@ -271,3 +246,257 @@ def ParticleFile(self, *args, **kwargs):
         """Wrapper method to initialise a :class:`parcels.particlefile.ParticleFile`
         object from the ParticleSet"""
         pass
+
+    def _set_particle_vector(self, name, value, indices=None):
+        """Set attributes of all particles to new values.
+
+        This is a fallback implementation, it might be slow.
+
+        :param name: Name of the attribute (str).
+        :param value: New value to set the attribute of the particles to.
+        :param indices: (Optional) only set the particles with these indices.
+                        Its length should be equal to the length of 'values'.
+                        If None, all particles are set.
+        """
+
+        if indices is None:
+            bool_indices = np.full(len(self), True)
+        else:
+            bool_indices = np.full(len(self), False)
+            bool_indices[indices] = True
+        for i, p in enumerate(self):
+            if bool_indices[i]:
+                setattr(p, name, value)
+
+    def _get_particle_vector(self, name, indices=None):
```

Comment:
still not really generalisable - I'm gonna make a counter-proposal here.

Reply:
That's not necessary. At the moment it is not used anywhere. I'll comment it out at least in my next commit.